pkg/controller/service/service_controller.go

```diff
@@ -636,9 +526,13 @@ func (s *ServiceController) nodeSyncLoop() {
 	glog.Infof("Detected change in list of current cluster nodes. New node set: %v",
 		nodeNames(newHosts))
 
-	// Try updating all services, and save the ones that fail to try again next
-	// round.
-	s.servicesToUpdate = s.cache.allServices()
+	// Try updating all services, and save the ones that fail to try again next round.
+	s.servicesToUpdate, err = s.serviceLister.List(labels.Everything())
```

Comment:
In the previous logic, it was guaranteed that EnsureLoadBalancer would be called before UpdateLoadBalancer for a new service. From what I see, this guarantee will no longer exist. For GCE, this will just result in an error and retries until the service syncs. Is it a problem for any other cloud?

Reply:
I don't see a problem with that approach. I'm guessing you don't have time to tackle #53462 for 1.10?

`s.servicesToUpdate` doesn't seem to be used outside this func. Can we make this into a local variable?

Reply:
> `s.servicesToUpdate` doesn't seem to be used outside this func. Can we make this into a local variable?

We definitely could clean that up in a different PR fairly easily. 

https://github.com/kubernetes/kubernetes/blob/bfdd94c6a0ea94cff5438c504c140c94b4c8e260/pkg/controller/service/service_controller.go#L194

>  I'm guessing you don't have time to tackle #53462 for 1.10?

Yeah, i'm not going to have time :/